src/mlpack/core/cereal/pointer_vector_variant_wrapper.hpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * @file core/cereal/pointer_vector_variant_wrapper.hpp
+ * @author Omar Shrit
+ *
+ * Implementation of a boost::variant wrapper to enable the serialization of
+ * the pointers inside boost variant in cereal
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CEREAL_POINTER_VECTOR_VARIANT_WRAPPER_HPP
+#define MLPACK_CORE_CEREAL_POINTER_VECTOR_VARIANT_WRAPPER_HPP
+
+#include "pointer_wrapper.hpp"
+#include "pointer_variant_wrapper.hpp"
+#include "pointer_vector_wrapper.hpp"
+
+namespace cereal {
+
+template<typename... VariantTypes>
+class pointer_vector_variant_wrapper;
+
+template<typename... VariantTypes>
+inline pointer_vector_variant_wrapper<VariantTypes...>
+make_vector_pointer_variant(std::vector<boost::variant<VariantTypes...>>& t)
+{
+  return pointer_vector_variant_wrapper<VariantTypes...>(t);
+}
+
+template<typename... VariantTypes>
+class pointer_vector_variant_wrapper
```

Comment:
Same comments here about the class naming and the location of the class documentation. :+1: